hs_tracking/views.py

```diff
@@ -4,9 +4,52 @@
 from django.views.generic import TemplateView
 from django.utils.decorators import method_decorator
 from django.contrib.auth.decorators import user_passes_test
-from django.http import HttpResponse
+from django.http import HttpResponse, HttpResponseRedirect
 
 from . import models as hs_tracking
+from .models import Session, Variable
+from .utils import get_std_log_fields
+
+import urlparse
+
+
+class AppLaunch(TemplateView):
+
+    def get(self, request, **kwargs):
+
+        # get the query parameters and remove the redirect url b/c
+        # we don't need to log this.
+        querydict = dict(request.GET)
+        url = querydict.pop('url')[0]
+
+        # log app launch details if user is logged in
+        if request.user.is_authenticated():
+
+            # get user session and standard fields
+            session = Session.objects.for_request(request, request.user)
+            fields = get_std_log_fields(request, session)
+
+            # parse the query and param portions of the url
+            purl = urlparse.urlparse(url)
+
+            # extract the app url args so they can be logged
+            app_args = urlparse.parse_qs(purl.query)
+
+            # update the log fields with the extracted request and url params
+            fields.update(querydict)
+            fields.update(app_args)
+
+            # clean up the formatting of the query and app arg dicts
+            # i.e. represent lists in csv format without brackets [ ]
+            # so that the log records don't need to be cleaned later.
+            fields.update(dict((k, ','.join(v)) for k, v in fields.iteritems()
+                          if type(v) == list))
+
+            # format and save the log message
+            msg = Variable.format_kwargs(**fields)
```

Comment:
I originally wrote this function to log any set of url parameters that are provided when an app is launched.   I'm not checking against a set of allowable fields so that we can easily extend the fields that we're logging.  Is this a bad approach?

Here is a sample url that the function takes:
```
href="/tracking/applaunch/?url=https%3A//apps.hydroshare.org/apps/hydroshare-gis/%3Fres_id%3D7a7de92941a044049a7b8ad09f4c75bb%26res_type%3DGenericResource;name=test"
```

The `get_std_log_fields` gets a set of fields that is consistent with all logged messages
```
fields = get_std_log_fields(request, session)
```

I then extend this set of fields with a more specific group of parameters that are passed into the function.  I'm not performing any filtering because these parameters are coded in the html templates (`title-section.html` and `genericresource.html`):

```
<a href="{% url 'tracking-applaunch' %}?url={{tool.url|urlencode}};name={{tool.title}}" target="_blank" >
```

and here is what's logged

```
res_type=GenericResource name=test user_email_domain=com user_type=Unspecified user_ip=192.168.56.1 res_id=7a7de92941a044049a7b8ad09f4c75bb
```